analog-in/maven/src/main/java/iotdk/example/AnalogIn.java

```diff
@@ -53,20 +51,25 @@ public static void checkRoot(){
 			System.out.println(message);
 		}
     }
-	public static void main(String[] args) {
-		
-		checkRoot();
 
+	public static void initPlatform(){
 		Platform platform = mraa.getPlatformType();
-		int pinNumber = 2;
 		if(platform.equals(Platform.INTEL_UP2)) {
 			if(USING_GROVE_PI_SHIELD) {
 				mraa.addSubplatform(Platform.GROVEPI, "0");
-				pinNumber = pinNumber + 512; // A2 Connector (512 offset needed for the shield)
+				pinNumber = 2 + 512; // A2 Connector (512 offset needed for the shield)
 			}
```

Comment:
This likely needs an else statement. The UP2 doesn't have built-in analog pins and needs the GrovePi+ shield for this sample.